packages/snap-preact/src/setupEvents.test.tsx

```diff
@@ -193,4 +207,67 @@ describe('setupEvents', () => {
 			jest.clearAllMocks();
 		});
 	});
+
+	describe('controller/updateRecs', () => {
+		it('can listen for controller/updateRecs event and invoked without parameters', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+
+			//no data passed at all
+			expect(() => {
+				eventManager.fire('controller/updateRecs');
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
+
+			jest.clearAllMocks();
+		});
+
+		it('can pass exact match controllerId', () => {
+			expect(makeSearchSpy).not.toHaveBeenCalled();
+			const controllerIds = 'recommend_similar_0';
+			expect(() => {
+				eventManager.fire('controller/updateRecs', { controllerIds });
+			}).not.toThrow();
+
+			expect(makeSearchSpy).toHaveBeenCalledTimes(1);
```

Comment:
How can we be sure the search was called on the controller we provide the `id` for?